src/Compiler/Checking/Expressions/CheckComputationExpressions.fs

```diff
@@ -1914,9 +1921,21 @@ let rec TryTranslateComputationExpression
                             mBind
                         )
 
+                    // Build the Using call that consumes the bound value
+                    // Reapply any optional type annotation to the resource identifier so errors are reported
+                    // at the annotation site without affecting the Bind parameter type.
+                    let resourceExpr =
+                        match typeAnnOpt with
+                        | Some(ty, mTy) ->
+                            // Apply the type annotation to the identifier expression to surface type errors on the annotation
+                            SynExpr.Typed(SynExpr.Ident ident, ty, unionRanges ident.idRange mTy)
+                        | None -> SynExpr.Ident ident
```

Comment:
@edgarfgp Could you explain this logic in greater detail, please?

There're very tricky details related to the types in computation expression builder methods, so I think it would be nice if everyone could follow your thought process here.

Reply:
@auduchinok My current attempt is to move the annotation off the Bind lambda parameter to the value we pass to Using so we don't force Bind type parameter(We let Bind pick the right input type from overloads by leaving the lambda parameter untyped.), but we still type-check the user’s annotation and report errors. But as Alex shared this wont be enough. My goal here is to find the minimal fix for this issue. Not to redesign the `ExprAsUseBang` case.